src/unix/fsevents.c

```diff
@@ -265,32 +265,36 @@ static void uv__fsevents_event_cb(ConstFSEventStreamRef streamRef,
 
         /* Skip forward slash */
         if (*path != '\0') {
+          /* Make sure that realpath actually named a directory */
+          if (*path != '/')
+            continue;
           path++;
           len--;
         }
       }
 
-#ifdef MAC_OS_X_VERSION_10_7
       /* Ignore events with path equal to directory itself */
-      if (len == 0)
-        continue;
-#else
-      if (len == 0 && (flags & kFSEventStreamEventFlagItemIsDir))
-        continue;
-#endif /* MAC_OS_X_VERSION_10_7 */
+      if (len == 0) {
+        if (flags & kFSEventStreamEventFlagItemIsDir)
+          continue;
+        /* Since we're using fsevents to watch the file itself,
+         * realpath == path, and we now need to get the basename of the file back
+         * (for commonality with other codepaths and platforms). */
+        while (len < handle->realpath_len && path[-1] != '/') {
```

Comment:
We got here by proving that handle->realpath is the same as path (since len == 0). In your example above, `len` would be 3, and we would not get here.